From 8139f988d8048b235cad67c8dc3393aea56d590d Mon Sep 17 00:00:00 2001
Message-Id: <8139f988d8048b235cad67c8dc3393aea56d590d.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Hong Liu <hong.liu@intel.com>
Date: Fri, 14 Nov 2014 11:17:27 +0800
Subject: [PATCH 31/34] REVERTME [VPG]: drm/i915: fix panic in
 i9xx_set_pipeconf

When resuming with HDMI connected, we got following panic.
During modesetting, we may get all 0 pipe masks, but valleyview adds
pipe 0 and pipe 2 to the prepare pipe mask, causing driver tries to do
crtc_enable on pipe 2 with uninitialized pipe_config, f.e pipe_bpp is 0,
this causes BUG() in i9xx_set_pipeconf().

REVERTME: This patch can be reverted once we figure out why we get all 0
pipe masks during resume with HDMI connected.

Issue: GMINL-3681
Change-Id: Iadfcfddd2e682047706aef5284070cfc0542c9d3
Signed-off-by: Hong Liu <hong.liu@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 7b57ebf..43ca052 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11949,6 +11949,9 @@ static int __intel_set_mode(struct drm_crtc *crtc,
 	intel_modeset_affected_pipes(crtc, &modeset_pipes,
 				     &prepare_pipes, &disable_pipes);
 
+	if (!(modeset_pipes | prepare_pipes | disable_pipes))
+		goto out;
+
 	*saved_mode = crtc->mode;
 
 	/* Hack: Because we don't (yet) support global modeset on multiple
-- 
1.7.9.5

